### ISSUE TITLE
KL divergence loss

### ISSUE THREAD

**Issue by taynoel84** [lucidrains/PaLM-rlhf-pytorch#38]
Hi, thanks for the great repo
I have a question,
In the function `masked_kl_div` of `ppo.py`, shouldnt the calculation be `prob1*(log(prob1) - log(prob2))`? 
The calculation in the code is a negative KL loss that is to be maximized instead of minimized (as assumed by the code).

**Comment by lucidrains** [comment]
@taynoel84 🤦 thank you for finding this error!

**Comment by dwyzzy** [comment]
Hi, thanks for the great work.
I also have a question about KL divergence loss.
In papers like [Learning to summarize from human feedback](https://arxiv.org/pdf/2009.01325.pdf), the KL item for human feedback policies seems to be the KL divergence between $\pi^{RL}$ and $\pi^{SFT}$, while in this repo the [code](https://github.com/lucidrains/PaLM-rlhf-pytorch/blob/ad001d87525b4c54870d2ea54bf03ae45a62ba24/palm_rlhf_pytorch/ppo.py#L508)
```
kl_div_loss = masked_kl_div(action_probs, old_action_probs, mask = action_masks) * self.kl_div_loss_weight
```
seems to be the KL divergence between $\pi^{new}$ and $\pi^{old}$.

Does there exist something wrong with the code, or have I made some mistakes?
Thank you.
